rust/main/hyperlane-core/src/rpc_clients/fallback.rs

```diff
@@ -195,6 +229,9 @@ where
                 let provider = &self.inner.providers[priority.index];
                 let resp = f(provider.clone()).await;
                 self.handle_stalled_provider(priority, provider).await;
+                if resp.is_err() {
+                    self.handle_failed_provider(idx).await;
+                }
```

Comment:
If a provider was deprioritised due to it stalled, we can reset the `last_failed_count` for it as well. Deprioritisation of such a provider due to failure is excessive, isn't it? 🤔